merkle/merkle_path.go

```diff
@@ -152,19 +164,59 @@ func pathFromNodeToRootAtSnapshot(node int64, level int, snapshot int64, maxBitL
 	return proof, nil
 }
 
-// subtreeDepth calculates the depth of a subtree, used at the right of the tree which
-// may not be completely populated
-func subtreeDepth(size int64, bits int) int {
-	for b := bitLen(size) - 1; b > bits; b-- {
-		size = size &^ (1 << uint(b))
+// lastNodeWritten determines if the last node is present in storage for a given Merkle tree size
+// and level in the tree (0 = leaves, increasing towards the root). This is determined by
+// examining the bits of the last valid leaf index in a tree of the specified size. Zero bits
+// indicate nodes that are not stored at that tree size.
+//
+// Example:
+//
+//             hash2
+//             /  \
+//            /    \
+//           /      \
+//          /        \
+//         /          \
+//        k            [X]
+//       / \             \
+//      /   \             \
+//     /     \             \
+//    g       h           [X]
+//   / \     / \         /
+//   a b     c d        e
+//   | |     | |        |
+//   d0 d1   d2 d3      d4
+//
+// Tree size = 5, last index = 4 in binary = 100, append 1 for leaves = 1001.
+// Reading down the RHS: present, not present, not present, present = 1001. So when
+// attempting to fetch the sibling of k the tree should be descended twice to fetch
+// 'e'.
+func lastNodeWritten(d, ts int64) bool {
```

Comment:
my brains keeps reading this as a func which checks that something's been written ok, rather than determining whether a node _should_ exist in the given place.
I'm not sure what an alternative name might be though...